nativelib/src/main/scala/scala/scalanative/native/Ptr.scala

```diff
@@ -1,142 +1,187 @@
-// ###sourceLocation(file: "/Users/Denys/.src/native/nativelib/src/main/scala/scala/scalanative/native/Ptr.scala.gyb", line: 1)
+// ###sourceLocation(file: "/home/denys/.src/native/nativelib/src/main/scala/scala/scalanative/native/Ptr.scala.gyb", line: 1)
 package scala.scalanative
 package native
 
-import runtime._
+import scala.runtime.BoxesRunTime._
+import scala.reflect.ClassTag
+import scalanative.runtime._
+import scalanative.runtime.Intrinsics._
+import scalanative.runtime.Boxes._
 
-/** The C `const T *` pointer. */
-final abstract class Ptr[T] {
+final class Ptr[T](private[scalanative] val rawptr: RawPtr) {
```

Comment:
@shadaj Here is a concrete example of what I meant in the discussion of #1420. `Ptr` type is a wrapper over `RawPtr`. User-facing API is safe from the boxing point of view and encapsulates raw unmanaged `RawPtr`. 

Similarly for `Word` boxing to work, `Word` itself could be a wrapper over `RawWord` which maps to a numeric type under the hood:
```
// .../native/Word.scala
final class Word(private[scalanative] val rawword: RawWord) {
  ...
  @inline def +(other: Word): Word =
     new Word(Intrinsics.addWord(this.rawword, other.rawword))
  ...
}

// .../runtime/RawWord.scala
final abstract class RawWord

// .../runtime/Intrinsics.scala
object Intrinsics {
  ...
  def addWord(l: RawWord, r: RawWord): RawWord = intrinsic
  ...
}
```

Reply:
Awesome, thanks for the example! The semester just started so still adjusting to the workload in my new schedule, but I'll try to update my PR to use this style sometime this week.